eng/pipelines/templates/stages/vmr-build.yml

```diff
@@ -35,6 +35,12 @@ parameters:
     name: NetCore-Public-XL
     demands: ImageOverride -equals Build.Ubuntu.1804.Amd64.Open
 
+variables:
+  isPR: $[eq(variables['Build.Reason'], 'PullRequest')]
+  isRelease: $[startsWith(variables['Build.SourceBranch'], 'refs/heads/release/')]
+  isInternalRelease: $[startsWith(variables['Build.SourceBranch'], 'refs/heads/internal/release/')]
```

Comment:
From the places of consumption, I think the code would be more readable if isRelease and isInternalRelease were combined into one variable.

Reply:
Azure DevOps has issues with these variables because they aren't compiled at runtime. To address this, I had to use [Build.DefinitionName](https://github.com/dotnet/installer/pull/17370/commits/aad69ec5a52c6b74d4f6e4dfd4edc2ecdf8e115b#diff-681076b03a7ed5a0cd4fca97e3de0303bb65604b0d20e05885f35332dd1a18ebR127) in order to distinguish between the pipelines. The changes are in [this commit](https://github.com/dotnet/installer/pull/17370/commits/aad69ec5a52c6b74d4f6e4dfd4edc2ecdf8e115b#).